vpr/src/route/rr_graph.xsd

```diff
@@ -0,0 +1,373 @@
+
+       For strings with a set of valid values (e.g. enumerations), the "type"
+       attribute should be the name of a xs:simpleType that defines an
+       xs:restriction / xs:enumeration.  This will be demostrated later.
+  -->
+  <xs:complexType name="channel">
+    <xs:attribute name="chan_width_max" type="xs:int" use="required"/>
+    <xs:attribute name="x_min" type="xs:int" use="required"/>
+    <xs:attribute name="y_min" type="xs:int" use="required"/>
+    <xs:attribute name="x_max" type="xs:int" use="required"/>
+    <xs:attribute name="y_max" type="xs:int" use="required"/>
+  </xs:complexType>
+
+  <xs:complexType name="x_list">
+    <!-- Here is an example of an unsigned int -->
+    <xs:attribute name="index" type="xs:unsignedInt" use="required"/>
+    <xs:attribute name="info" type="xs:int" use="required"/>
+  </xs:complexType>
+
+  <xs:complexType name="y_list">
+    <xs:attribute name="index" type="xs:unsignedInt" use="required"/>
+    <xs:attribute name="info" type="xs:int" use="required"/>
+  </xs:complexType>
+
+  <!-- This an example tag definition that is made up of child tags.
+       In this case there are three types of child tags.  The "channel" tag
+       has one instance, which is the default.
+
+       Both x_list and y_list can have multiple instances, so
+       maxOccurs="unbounded" is added to indicate that more than 1 are allowed,
+       and that there is no upper bound.
+
+       In all three cases, the type string refers to a xs:complexType that was
+       defined above.
+  -->
+  <xs:complexType name="channels">
+    <xs:sequence>
+      <xs:element name="channel" type="channel"/>
+      <xs:element name="x_list" type="x_list" maxOccurs="unbounded"/>
+      <xs:element name="y_list" type="y_list" maxOccurs="unbounded"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <!-- This is an example string enumeration type definition.
+       The type of the enumeration is a string, hence the base="xs:string".
+       Inside of the xs:restriction are the allowed values that this enumeration
+       can take.
+    -->
+  <xs:simpleType name="switch_type">
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="mux"/>
+      <xs:enumeration value="tristate"/>
+      <xs:enumeration value="pass_gate"/>
+      <xs:enumeration value="short"/>
+      <xs:enumeration value="buffer"/>
+    </xs:restriction>
+  </xs:simpleType>
+
+  <xs:complexType name="timing">
+    <!-- Here is an example of an optional attribute.  In the previous
+         definitions use="required" was specified indicating that the
+         attribute must be present.  In this case, these attributes are not
+         required, so they do not specified use="required".
+    -->
+    <xs:attribute name="R" type="xs:float"/>
+    <xs:attribute name="Cin" type="xs:float"/>
+    <xs:attribute name="Cinternal" type="xs:float"/>
+    <xs:attribute name="Cout" type="xs:float"/>
+    <xs:attribute name="Tdel" type="xs:float"/>
```

Comment:
Should probably have the penalty_cost attribute

Reply:
Yep, I'll fix that now.

Reply:
Fixed.